Update emoji list to emoji v15.0.

lushangkan opened the pull request:
## Context

Update emoji list to emoji version.
Add option to customize emoji list.

## Summary

Update emoji list to emoji v15.0.
Change emoji grouping to official grouping.
Add option to customize emoji list.
Change default group icon to lucide icon.
Modify Header Padding to accommodate new grouping size.
Modify Header flex option to fix off-center icons.
Updating npm dependencies.
Fix some bugs.
And more...

## To-do

- [ ] Due to the change in emoji grouping, users may need to be reminded of the change in the documentation in the future.

## User-facing changes

Too much changes. Comment it if you need.

## Checklist

<!-- Check these after PR creation -->

- [x] I have tested this code to the best of my abilities
- [x] I have added documentation where necessary

delowardev commented:
Hi @lushangkan Thank you for the PR; I will review the changes and merge it as soon as possible.

lushangkan commented:
Seems like I didn't handle skin tones well, I'll fix that later

lushangkan commented:
Fixed unsupported skin color issue, [zwj_emoji](https://[unicode.org/Public/emoji/15.0/emoji-zwj-sequences.txt](https://unicode.org/Public/emoji/15.0/emoji-zwj-sequences.txt)) not supported for now (except Other group)

delowardev reviewed (changes requested):
Hi @lushangkan, thanks again for your incredible effort on this. 

PR looks good, just two minor changes needed:

1. You may want to update the emoji cdn URL as well in `constant.ts#L3`, the new src should be ```https://cdn.jsdelivr.net/npm/emoji-datasource-apple@15.0.1/img/apple/64/```

https://github.com/delowardev/vue3-emoji-picker/blob/c430646adea6727f0b31349ec33629b72a7715f2/src/constant.ts#L3-L4

2. In emojis.json#10461, could you please update `2642-fe0f` to `2641-fe0f`? I noticed the `male sign` emoji doesn't show with the previous code for some reason. 

https://github.com/delowardev/vue3-emoji-picker/blob/c430646adea6727f0b31349ec33629b72a7715f2/src/data/emojis.json#L10461

lushangkan commented:
> Hi @lushangkan, thanks again for your incredible effort on this.
> 
> PR looks good, just two minor changes needed:
> 
> 1. You may want to update the emoji cdn URL as well in `constant.ts#L3`, the new src should be `https://cdn.jsdelivr.net/npm/emoji-datasource-apple@15.0.1/img/apple/64/`
> 
> https://github.com/delowardev/vue3-emoji-picker/blob/c430646adea6727f0b31349ec33629b72a7715f2/src/constant.ts#L3-L4
> 
> 2. In emojis.json#10461, could you please update `2642-fe0f` to `2641-fe0f`? I noticed the `male sign` emoji doesn't show with the previous code for some reason.
> 
> https://github.com/delowardev/vue3-emoji-picker/blob/c430646adea6727f0b31349ec33629b72a7715f2/src/data/emojis.json#L10461

Thank you for checking it out!
I've noticed some new issues in the meantime, I'll get back to you when I'm done fixing them.
Also, emoji added a lot of [zwj emoji](https://[unicode.org/Public/emoji/15.0/emoji-zwj-sequences.txt](https://unicode.org/Public/emoji/15.0/emoji-zwj-sequences.txt)) to switch the gender of the character within the emoji, etc., do you have plans to adapt it?

delowardev commented:
I think zwj emoji will be a good feature, I will need to check the possibilities.

lushangkan commented:
> 2\. 2641-fe0f

Is the 2641 the ♁ symbol you are referring to? The symbol is [earth symbol] (1)，[deprecated in Unicode 1.1](2), I switched 2642 to 2641 in my device (Windows 11) and it shows up as ![image](https://github.com/delowardev/vue3-emoji-picker/assets/41252325/9f734ebc-7bc6-41dd-b1e9-5037fbff5d8f), and I didn't find this icon in the official list of Unicode either

[1](https://www.emojiall.com/en/emoji/%E2%99%81)
[2](https://emojipedia.org/earth)

lushangkan commented:
> > 2. 2641-fe0f
> 
> Is the 2641 the ♁ symbol you are referring to? The symbol is [earth symbol](1)，[deprecated in Unicode 1.1](2), I switched 2642 to 2641 in my device (Windows 11) and it shows up as ![image](https://user-images.githubusercontent.com/41252325/261856451-9f734ebc-7bc6-41dd-b1e9-5037fbff5d8f.png), and I didn't find this icon in the official list of Unicode either
> 
> [1](https://www.emojiall.com/en/emoji/%E2%99%81) [2](https://emojipedia.org/earth)



> Hi @lushangkan, thanks again for your incredible effort on this.
> 
> PR looks good, just two minor changes needed:
> 
> 1. You may want to update the emoji cdn URL as well in `constant.ts#L3`, the new src should be `https://cdn.jsdelivr.net/npm/emoji-datasource-apple@15.0.1/img/apple/64/`
> 
> https://github.com/delowardev/vue3-emoji-picker/blob/c430646adea6727f0b31349ec33629b72a7715f2/src/constant.ts#L3-L4
> 
> 2. In emojis.json#10461, could you please update `2642-fe0f` to `2641-fe0f`? I noticed the `male sign` emoji doesn't show with the previous code for some reason.
> 
> https://github.com/delowardev/vue3-emoji-picker/blob/c430646adea6727f0b31349ec33629b72a7715f2/src/data/emojis.json#L10461

Is the 2641 the ♁ symbol you are referring to? The symbol is [earth symbol](https://www.emojiall.com/en/emoji/%E2%99%81), [deprecated in Unicode 1.1](https://emojipedia.org/earth), I switched 2642 to 2641 in my device (Windows 11) and it shows up as ![image](https://github.com/delowardev/vue3-emoji-picker/assets/41252325/9f734ebc-7bc6-41dd-b1e9-5037fbff5d8f), and I didn't find this icon in the official list of Unicode either

lushangkan commented:
> Hi @lushangkan, thanks again for your incredible effort on this.
> 
> PR looks good, just two minor changes needed:
> 
> 1. You may want to update the emoji cdn URL as well in `constant.ts#L3`, the new src should be `https://cdn.jsdelivr.net/npm/emoji-datasource-apple@15.0.1/img/apple/64/`
> 
> https://github.com/delowardev/vue3-emoji-picker/blob/c430646adea6727f0b31349ec33629b72a7715f2/src/constant.ts#L3-L4
> 
> 2. In emojis.json#10461, could you please update `2642-fe0f` to `2641-fe0f`? I noticed the `male sign` emoji doesn't show with the previous code for some reason.
> 
> https://github.com/delowardev/vue3-emoji-picker/blob/c430646adea6727f0b31349ec33629b72a7715f2/src/data/emojis.json#L10461

I see the request https://fastly.jsdelivr.net/npm/emoji-datasource-apple@15.0.1/img/apple/64/2642-fe0f.png failed message in the console, which could be a problem with the emoji-datasource-apple repository
